File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    
    lst = []
    for s in score:
        if len(lst) < k:
            lst.append(s)
        else:
            if s > min(lst):
                lst.remove(min(lst))
                lst.append(s)
                
        answer.append(min(lst))
    
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    
    let honorList = [];
    for (let num of score) {
        if (honorList.length < k) honorList.push(num);
        else {
            if (num >= honorList[0]) {
                honorList.shift();
                honorList.push(num);
            }
        }
        honorList.sort((a, b) => a - b);
        answer.push(honorList[0]);
    }
    
    return answer;
}